sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs

```diff
@@ -106,7 +125,25 @@ void AddCustomerPolicies(HttpPipelinePosition position)
 
             policies.RemoveAll(static policy => policy == null);
 
-            return new HttpPipeline(options.Transport,
+            // Override the provided Transport with the provided transport options if the transport has not been set after default construction and options are not null.
+            HttpPipelineTransport transport = options.Transport;
+            if (defaultTransportOptions != null)
+            {
+                if (options.IsCustomTransportSet)
+                {
+                    if (AzureCoreEventSource.Singleton.IsEnabled())
+                    {
+                        // Log that we were unable to override the custom transport
+                        AzureCoreEventSource.Singleton.PipelineTransportOptionsNotApplied(options?.GetType().FullName ?? String.Empty);
+                    }
+                }
+                else
+                {
+                    transport = HttpPipelineTransport.Create(defaultTransportOptions);
+                }
+            }
+
+            return new DisposableHttpPipeline(transport,
```

Comment:
If someones manages to downcast this to IDisposable are they going to dispose a shared transport?

Reply:
Great callout - I was thinking we could protect against this in the DisposeInternal implementation to compare this with Shared and only dispose if so.